pkg/scheduler/apis/config/validation/validation_pluginargs.go

```diff
@@ -311,15 +311,27 @@ func ValidateNodeAffinityArgs(path *field.Path, args *config.NodeAffinityArgs) e
 	return errors.Flatten(errors.NewAggregate(errs))
 }
 
+// VolumeBindingArgsValidationOptions contains the different settings for validation.
+type VolumeBindingArgsValidationOptions struct {
```

Comment:
What's the motivation for introducing this? If it's to eliminate the dependency on `utilfeature`, L322 still depends on it; if not, the old logic that directly checks the feature gate seems more straightforward.

Reply:
This was inspired by my initial recommendation. There are 2 options:

- Keep the Options but populate it outside of this package.
- Leave the old logic (we probably don't care enough about the dependency). Just don't import `k8s.io/kubernetes/pkg/scheduler/framework/plugins/feature` here.

Reply:
The 2nd option sounds neat to me.

Reply:
hi, @Huang-Wei 

I'd prefer keeping this option struct. It's required if we want to cut the dependency on the feature gates (typically the global `utilfeature.DefaultFeatureGate`). Without this option struct, we need to pass `fts.Features` to `utilfeature.DefaultFeatureGate`, because we need to call validation function in `volumebinding` plugin if the validation function check the features in `utilfeature.DefaultFeatureGate`. This will be weird.

Reply:
The above comment is not for pkg/scheduler/framework/plugins/volumebinding/volume_binding.go - that refactoring is absolutely fine.

My point is it doesn't seem necessary to introduce the option in validation logic. There are no differences in terms of dependency as both depend on utilfeature, isn't it?

So I'd prefer to revert the changes in this file.

```diff
diff --git a/pkg/scheduler/apis/config/validation/validation_pluginargs.go b/pkg/scheduler/apis/config/validation/validation_pluginargs.go
index 70fdabe20b1..3202841c579 100644
--- a/pkg/scheduler/apis/config/validation/validation_pluginargs.go
+++ b/pkg/scheduler/apis/config/validation/validation_pluginargs.go
@@ -311,27 +311,15 @@ func ValidateNodeAffinityArgs(path *field.Path, args *config.NodeAffinityArgs) e
 	return errors.Flatten(errors.NewAggregate(errs))
 }
 
-// VolumeBindingArgsValidationOptions contains the different settings for validation.
-type VolumeBindingArgsValidationOptions struct {
-	AllowVolumeCapacityPriority bool
-}
-
 // ValidateVolumeBindingArgs validates that VolumeBindingArgs are set correctly.
 func ValidateVolumeBindingArgs(path *field.Path, args *config.VolumeBindingArgs) error {
-	return ValidateVolumeBindingArgsWithOptions(path, args, VolumeBindingArgsValidationOptions{
-		AllowVolumeCapacityPriority: utilfeature.DefaultFeatureGate.Enabled(features.VolumeCapacityPriority),
-	})
-}
-
-// ValidateVolumeBindingArgs validates that VolumeBindingArgs with scheduler features.
-func ValidateVolumeBindingArgsWithOptions(path *field.Path, args *config.VolumeBindingArgs, opts VolumeBindingArgsValidationOptions) error {
 	var allErrs field.ErrorList
 
 	if args.BindTimeoutSeconds < 0 {
 		allErrs = append(allErrs, field.Invalid(path.Child("bindTimeoutSeconds"), args.BindTimeoutSeconds, "invalid BindTimeoutSeconds, should not be a negative value"))
 	}
 
-	if opts.AllowVolumeCapacityPriority {
+	if utilfeature.DefaultFeatureGate.Enabled(features.VolumeCapacityPriority) {
 		allErrs = append(allErrs, validateFunctionShape(args.Shape, path.Child("shape"))...)
 	} else if args.Shape != nil {
 		// When the feature is off, return an error if the config is not nil.

```

Reply:
The new option struct is introduced because we need to pass the `fts.Features` to the validation function. Especially in unit testing, if we don't the test will fail. 

However, the `fts.Features` is populated from the global `utilfeature.DefaultFeatureGate` in `kube-scheduler`. This is the place why I think it will be weird.

Reply:
I think the struct is fine. We could try to get rid of the original `ValidateVolumeBindingArgs` as well, but we would need to pass the options to `ValidateKubeSchedulerConfiguration` or something like that.

Reply:
> The new option struct is introduced because we need to pass the fts.Features to the validation function. Especially in unit testing, if we don't the test will fail.

That's fair.

> We could try to get rid of the original `ValidateVolumeBindingArgs` as well

That sounds great so we don't need to build another options struct - just use `plugins/../feature.go#Features`.  Also, we would change the validation function signature to `func(path, args, options)` and then parse the signature properly. Let's pursue it in a follow-up.

Reply:
> Let's pursue it in a follow-up.

Hmm... it seems we cannot achieve it as the `fts` can only be obtained from New(), which happens after validation (Config()).